### PR TITLE
support for drop-in unit files

### PR DESCRIPTION
**Pull request by rj667**
Hi, this should fix issue #15. We run this in production on 300+ systemd servers.

### CODE REVIEW

File: manifests/unit_file.pp [hunk 1]
```diff
@@ -9,6 +9,25 @@
...
 
+  if $title =~ /^([^\/]+)\// {
+    $subdir = $1
+    if $subdir !~ /\.d$/ {
+      fail('drop-in unit dirs must have suffix .d')
+    }
+    if $title !~ /\.conf$/ {
+      fail('drop-in unit files must have suffix .conf')
+    }
+    file { "${path}/${subdir}":
```
Comment (raphink) [inline manifests/unit_file.pp]: This will fail if you try to manage several files in the same subdir.

Comment (rj667) [inline manifests/unit_file.pp]: You are right. We don't do that, so I hadn't noticed. We do override multiple configuration options of a single service, but we put those in a single drop-in unit file. I wonder if there is an actual use case for multiple drop-in unit files for the same service but that is a different discussion.
That situation could be solved by replacing the file{} with an exec{} that does "mkdir -p". Would you like me to do that or would you rather reject the whole patch?

Comment (salekseev) [inline manifests/unit_file.pp]: +1

Comment (raphink) [inline manifests/unit_file.pp]: @rj667 I'd rather avoid an exec resource. If you have to take that road, I'd rather you use `ensure_directory()`

File: manifests/unit_file.pp [hunk 2]
```diff
@@ -9,6 +9,25 @@
 ) {
   include ::systemd
 
+  if $title =~ /^([^\/]+)\// {
```
Comment (bastelfreak) [inline manifests/unit_file.pp]: I think this whole logic can be simplified by using a proper datatype for the parameter?

**Comment by bastelfreak** [comment]
Hi @rj667, are you still interested in this PR? If so, can you please rebase it against our current master?

File: manifests/unit_file.pp (hunk 2 above)
Comment (ekohl) [reply manifests/unit_file.pp]: AFAIK titles can't have data types.

Comment (bastelfreak) [reply manifests/unit_file.pp]: Ah right, not in the params section I think. But we can still use assert_type()?

**Comment by kenyon** [comment]
This should be closed due to #39.

**Pull request closed without merging**